File: solutions/problem-6/solution.js
class Solution {
  // num is the number that you want to calculate the sums to
  constructor(num) {
    this.num = num;
  }
  
  sumOfSquares() {
    let sum = 0;
    for (let i = 1; i <= this.num; i++) {
      sum += Math.pow(i, 2);
    }
    return sum;
  }
  
  squaresOfSums() {
    let sum = 0;
    for (let i = 1; i <= this.num; i++) {
      sum += i;
    }
    return Math.pow(sum, 2);
  }

  solve() {
    return this.squaresOfSums() - this.sumOfSquares();
  }

}

const solution = new Solution(100);
solution.solve();

File: solutions/problem-6/solution.java
public class problem6 {
    static int squareSum(int num){
        num=(num * (num + 1) * (2 * num + 1)) / 6;
        return num;
    }
    static int square_sum(int n){
        n=(n * (n + 1)) / 2;  //sum of numbers
        n=n * n;  //square of num
        return n;
    }
    public static void main(String[] args) {
        System.out.println(Math.abs(square_sum(10) - squareSum(10)));//difference
    }
}
